test/e2e/token_inactivity_timeout_test.go

```diff
@@ -0,0 +1,233 @@
+	}
+
+	longLivedAccessToken := &oauthapi.OAuthAccessToken{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "Wf0AA1DphiD1FFLQV7BQIsesaaba-15RoSPZ3SvvXXT",
+		},
+		ClientName:               oauthClientName,
+		ExpiresIn:                defaultAccessTokenMaxAgeSeconds,
+		Scopes:                   []string{"user:full"},
+		RedirectURI:              redirectURIs[0],
+		UserName:                 userName,
+		UserUID:                  string(uid),
+		AuthorizeToken:           "mJOQ7Es5l9V7WYDl0bvl3E_hRjnJ21ZZxXH6YZj3yeT",
+		InactivityTimeoutSeconds: 300,
+	}
+
+	for _, accessToken := range []*oauthapi.OAuthAccessToken{shortLivedOAuthAccessToken, longLivedAccessToken} {
+		_, err = oauthClientClient.OAuthAccessTokens().Create(context.Background(), accessToken, metav1.CreateOptions{})
+		require.NoError(t, err)
+		defer func(name string) {
+			if err := oauthClientClient.OAuthAccessTokens().Delete(context.Background(), name, metav1.DeleteOptions{}); err != nil {
+				t.Logf("%v", err)
+			}
+		}(accessToken.Name)
+	}
+
+	httpTransport := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).DialContext,
+		TLSHandshakeTimeout: 10 * time.Second,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+
+	req, err := http.NewRequest(http.MethodGet, kubeConfig.Host+"/apis/user.openshift.io/v1/users/~", &bytes.Buffer{})
+	require.NoError(t, err)
+	req.Header.Set("Authorization", "Bearer "+shortLivedOAuthAccessToken.Name)
+
+	resp, err := httpTransport.RoundTrip(req)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		t.Fatalf("short lived access token should work. Recieved status %v", resp.Status)
+	}
+
+	// Wait for token to expire.
+	fmt.Println("Waiting for short lived access token to expire due to inactivity")
+	// Nanosecond delta make sure that test does not hit edge cases.
+	time.Sleep(300*time.Second + time.Nanosecond)
+
+	resp, err = httpTransport.RoundTrip(req)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		t.Fatalf("short lived access token should not work. Recieved status %v", resp.Status)
+	}
+
+	updateOAuthServerClusterConfigSpec(t, configClient, nil)
+
+	// Wait for kube-apiserver to start the new rollout
+	test.WaitForKubeAPIServerStartProgressing(t, configClient)
+
+	// Wait until all kube-apiserver pods are restarted with new config.
+	test.WaitForKubeAPIServerClusterOperatorAvailableNotProgressingNotDegraded(t, configClient)
+
+	req.Header.Set("Authorization", "Bearer "+longLivedAccessToken.Name)
+
+	resp, err = httpTransport.RoundTrip(req)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		t.Fatalf("long lived access token should still work. Recieved status %v", resp.Status)
+	}
+
+	fmt.Println("wait to check if long lived token works after inactivity timeout configured previously")
```

Comment:
you would need another short lived token here IMO. Checking the long lived one does not make much sense. It won't expire in any case.

Reply:
is it the name that is causing confusion? the objective here is to that token is still valid after inactive period even with timeout validator disabled. any token that lives more than this duration will work in this case

Reply:
Am a little confused. We document in the API https://github.com/openshift/api/blob/master/config/v1/types_oauth.go#L55 that tokens expire no-matter-what after the configured timeout. How does the set timeout value in the access token play into this? From the API docs of `TokenConfig.AccessTokenInactivityTimeout`, even the long-lived token should expire. What do I miss?

Reply:
timeout validator can expire token only if [inactivity timeout is set on the token.](https://github.com/openshift/kubernetes/blob/master/openshift-kube-apiserver/authentication/oauth/timeoutvalidator.go#L81-L92) As, oauth-server issues tokens with this timeout set, tokens issued by it will expire after inactive period. So, do you think that tokens created manually without this value must also expire?